armi/reactor/converters/axialExpansionChanger/axialExpansionChanger.py

```diff
@@ -331,53 +343,104 @@ def axiallyExpandAssembly(self):
             top-most Block is is then updated to reflect any expansion/contraction.
         """
         mesh = [0.0]
-        numOfBlocks = self.linked.a.countBlocksWithFlags()
         runLog.debug(
             "Printing component expansion information (growth percentage and 'target component')"
             f"for each block in assembly {self.linked.a}."
         )
+        # expand all of the components
```

Comment:
None blocking thought for you.

This is a LONG method. It is both a lot read as a reviewer (or ARMI dev) and it is just a lot of complicated logic.  Simplifying either of those would help.

But this is a pretty complicated system you've built. I cannot verify right now that this method should work; there's a lot here.

Reply:
Sure, if you self-assign that ticket I'm okay with it.

Reply:
> Ticket?

https://github.com/terrapower/armi/issues/2261